hack/cleanup/cleanup_oapi.py

```diff
@@ -0,0 +1,151 @@
+    parser.add_argument('--imageId', action='store', type=str, help='Outscale Image Id',)
+    parser.add_argument('--imageName', action='store', type=str, help='Outscale Image Name',)
+    parser.add_argument('--imageNameFilterPath', action='store', type=str, help='file path of Outscale Image Name to keep',)  
+    parser.add_argument('--imageNamePattern', action='store', type=str, help='Outscale Image Name Pattern',)
+    return parser.parse_args()
+
+# checkImageIdAttached check imageId exist
+def checkImageIdAttached(gateway, imageId, accountOwnerImageId):
+    """Check Image Id"""
+    vms = gateway.ReadVms().get('Vms', [])
+    vmsImageId = [vm["ImageId"] for vm in vms if 'ReservationId' in vm]
+    for vmImageId in vmsImageId:
+        if vmImageId == imageId:
+            print("Get Vm Image {}".format(imageId))
+            return True
+    return False
+
+# deleteImageId delete imageId
+def deleteImageId(gateway, imageId, accountOwnerImageId):
+    """Delete Image Id """
+    res = gateway.DeleteImage(ImageId=imageId)
+    if "Errors" in res:
+        print("Can not delete Image {} because {}".format(imageId, res))
+    print("Delete ImageId {}".format(imageId))
+
+# getImageId retrieve imageId        
+def getImageId(gateway, imageName, accountOwnerImageId):
+    """Get Image Id"""
+    for image in gateway.ReadImages(Filters={'ImageNames': [imageName], 'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        imageId = image['ImageId']
+        print("Get ImageId {}".format(imageId))
+        return imageId
+
+# checkImageIdExist check imageId exist
+def checkImageIdExist(gateway, imageId, accountOwnerImageId):
+    """Check Image Id exist"""
+    for image in gateway.ReadImages(Filters={'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        if image['ImageId'] == imageId:
+            print("ImageId {} does exist".format(imageId))
+            return True
+    print("ImageId {} does not exist".format(imageId))
+    return False
+    
+# deleteImage delete image.
+def deleteImage(gateway, accountOwnerImageId, daysCount, imageNamePattern, imageNameFilter):
+    """Get Image Date"""
+    for image in gateway.ReadImages(Filters={'AccountIds': [accountOwnerImageId]}).get('Images', []):
+        imageId = image['ImageId']
+        creationDate = image['CreationDate']
+        imageName = image['ImageName']
+        if imageNamePattern != "" and imageNamePattern in imageName and imageName not in imageNameFilter:
+            print("Image {} match imagePattern {}".format(imageName, imageNamePattern))
+            print("Get ImageName {} with ImageId {} in {}".format(imageName, imageId, creationDate))
+            checkImageToBeDelete(gateway, accountOwnerImageId, creationDate, imageId, daysCount)
+        else:    
+            print("ImageName {} will be keeped".format(imageName)) 
+
+# deleteUnattachedImage delete unattached image
+def deleteUnattachedImage(gateway, imageId, accountOwnerImageId):
```

Comment:
I think this is not necessary because again we can only look at which images are beeing used in our account not all accounts